src/server/server_impl.go

```diff
@@ -115,49 +115,51 @@ func (server *server) GrpcServer() *grpc.Server {
 func (server *server) Start() {
 	go func() {
 		addr := fmt.Sprintf(":%d", server.debugPort)
-		logger.Warnf("Listening for debug on '%s'", addr)
-		var err error
-		server.listenerMu.Lock()
-		server.debugListener.listener, err = reuseport.Listen("tcp", addr)
-		server.listenerMu.Unlock()
-
+		logger.Infof("Listening for debug on '%s'", addr)
+		lis, err := reuseport.Listen("tcp", addr)
 		if err != nil {
 			logger.Errorf("Failed to open debug HTTP listener: '%+v'", err)
 			return
 		}
-		err = http.Serve(server.debugListener.listener, server.debugListener.debugMux)
-		logger.Infof("Failed to start debug server '%+v'", err)
+		server.listenerMu.Lock()
+		server.debugListener.listener = lis
+		server.listenerMu.Unlock()
+
+		if err := http.Serve(server.debugListener.listener, server.debugListener.debugMux); err != nil {
+			logger.Fatalf("Failed to start debug server: '%+v'", err)
+		}
 	}()
 
 	go server.startGrpc()
 
 	server.handleGracefulShutdown()
 
 	addr := fmt.Sprintf(":%d", server.port)
-	logger.Warnf("Listening for HTTP on '%s'", addr)
-	list, err := reuseport.Listen("tcp", addr)
+	logger.Infof("Listening for HTTP on '%s'", addr)
+	lis, err := reuseport.Listen("tcp", addr)
 	if err != nil {
 		logger.Fatalf("Failed to open HTTP listener: '%+v'", err)
 	}
 	srv := &http.Server{Handler: server.router}
 	server.listenerMu.Lock()
 	server.httpServer = srv
 	server.listenerMu.Unlock()
-	err = srv.Serve(list)
 
-	if err != http.ErrServerClosed {
-		logger.Fatal(err)
+	if err := srv.Serve(lis); err != nil {
+		logger.Fatalf("Failed to start HTTP server: '%+v'", err)
```

Comment:
There are cases in which we do a graceful shutdown and we don't want to print the fatal and exit. This is actually the exit case, so I think you want the code that was there before, and I think you want to replicate this above and below also? cc @dweitzman

Reply:
💡 I get it now, makes perfect sense. I can correct that.

Reply:
Looks good. Thanks